src/Servers/Kestrel/test/InMemory.FunctionalTests/ResponseTests.cs

```diff
@@ -3819,19 +3822,17 @@ public async Task ResponseGetMemoryAndStartAsyncAdvanceThrows()
         }
 
         [Fact]
-        public async Task ResponseCompleteGetMemoryAdvanceInLoopDoesNotThrow()
+        public async Task ResponseCompleteGetMemoryDoesThrow()
         {
-            await using (var server = new TestServer(async httpContext =>
-            {
+            InvalidOperationException writeEx = null;
 
+            await using (var server = new TestServer(httpContext =>
+            {
                 httpContext.Response.BodyWriter.Complete();
-                for (var i = 0; i < 5; i++)
-                {
-                    var memory = httpContext.Response.BodyWriter.GetMemory(); // Shouldn't throw
-                    httpContext.Response.BodyWriter.Advance(memory.Length);
-                }
 
-                await Task.CompletedTask;
+                writeEx = Assert.Throws<InvalidOperationException>(() => httpContext.Response.BodyWriter.GetMemory());
```

Comment:
This part confuses me. I thought we wanted to have the same behavior as streams where calling WriteAsync after Complete noops. However, I  think the difference is that the noop behavior only occurs when the connection was aborted or the streams go away. Hence why you added the fake memory to HTTP2?

Reply:
We want the same behavior as Streams where aborting http connection or http/2 stream makes writing no-op. If the application explicitly completes the PipeWriter, that a different story entirely.